pkg/prometheus/resource_selector.go

```diff
@@ -120,7 +120,7 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 	res := make(map[string]*monitoringv1.ServiceMonitor, len(serviceMonitors))
 	for namespaceAndName, sm := range serviceMonitors {
 		var err error
-		rejectFn := func(_ *monitoringv1.ServiceMonitor, err error) {
```

Comment:
So I'm not sure why there this was changed to `_` as it was not when I initially did my first PR. It was named `servicemonitor` first and the issue is that the sm variable defined line 121 was used instead so I renamed it to `sm`. 

I'm not sure there is a reason to even have the servicemonitor paramater in the reject function as the code always use the loop value above but if there is, the function name should shadow the loop variable and if there is none, maybe the variable should just be removed in all resources (probes, podmonitors ...).

What do you think?

Reply:
it might a side effect of #6384?